pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py

```diff
@@ -1434,3 +1434,92 @@ def test_cross_repo_search_index_on_distribution_with_repository_and_deprecation
 
     # did it get properly marked as deprecated?
     assert resp.data[0].is_deprecated, resp
+
+
+def test_cross_repo_search_index_with_updated_namespace_metadata(
+    ansible_collection_deprecations_api_client,
+    ansible_distro_api_client,
+    ansible_namespaces_api_client,
+    ansible_repo_api_client,
+    ansible_repo_version_api_client,
+    build_and_upload_collection,
+    galaxy_v3_collections_api_client,
+    galaxy_v3_default_search_api_client,
+    galaxy_v3_plugin_namespaces_api_client,
+    gen_object_with_cleanup,
+    monitor_task,
+    random_image_factory,
+):
+    """Make sure namespace metdata updates are reflected in the index."""
+
+    # make a repo
+    pulp_repo = gen_object_with_cleanup(ansible_repo_api_client, {"name": str(uuid.uuid4())})
+
+    # make a distro that points at the repository
+    distro = gen_object_with_cleanup(
+        ansible_distro_api_client,
+        {
+            "name": pulp_repo.name,
+            "base_path": pulp_repo.name,
+            "repository": pulp_repo.pulp_href,
+        },
+    )
+    dist_id = distro.pulp_href.split("/")[-2]
+
+    # define col namespace
+    namespace_name = randstr()
+
+    # define col name
+    collection_name = randstr()
+
+    # make namespace metadata
+    kwargs = {"path": distro.base_path, "distro_base_path": distro.base_path}
+    task = galaxy_v3_plugin_namespaces_api_client.create(
+        name=namespace_name, description="hello", company="Testing Co.", **kwargs
+    )
+    result = monitor_task(task.task)
+    namespace_href = result.created_resources[1]
```

Comment:
I'm still unsure whether that order is stable.
(Reminds me that i want tasks to be able to actually return something.)

Reply:
I tried to use gen_object_with_cleanup, but something about that api client or maybe the async task is incompatible.

Reply:
Hah, no a task with two returned entitites is not compatible with that fixture.

Reply:
You can still use the add_to_cleanup fixture though.